### PR TITLE
 "Remove location" button for new uploads, and also remove exif data

### PR DESCRIPTION
**Pull request by shankarpriyank**
Fixed removal of exif data, I could not find paco's repository in the list of main repository hence I selected the main to be the base branch.
Adds work on top on #5262

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java [hunk 1]
```diff
@@ -540,7 +542,27 @@ public void editLocation(final String latitude, final String longitude, final do
...
+
+            for (String tag : exifTags) {
+                sourceExif.setAttribute(tag, null);
+            }
+            Toast.makeText(getContext(), "Location Removed", Toast.LENGTH_LONG).show();
+
+        } catch (IOException e) {
+           Timber.d(e);
+            Toast.makeText(getContext(), "Location Could Not be Removed", Toast.LENGTH_LONG).show();
+
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java:562]: Would you mind removing the extra blank lines here?

File: app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java [hunk 2]
```diff
@@ -540,7 +542,27 @@ public void editLocation(final String latitude, final String longitude, final do
...
+
+        }
+
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java:565]: and here

File: app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java [hunk 3]
```diff
@@ -540,7 +542,27 @@ public void editLocation(final String latitude, final String longitude, final do
...
+            };
+
+            for (String tag : exifTags) {
+                sourceExif.setAttribute(tag, null);
+            }
+            Toast.makeText(getContext(), "Location Removed", Toast.LENGTH_LONG).show();
+
+        } catch (IOException e) {
+           Timber.d(e);
+            Toast.makeText(getContext(), "Location Could Not be Removed", Toast.LENGTH_LONG).show();
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java:561]: indentation seems a bit off?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

**Comment by shankarpriyank** [comment]
Fixed
I think adding a linter would help save a lot of time I could use some help with #5198

**Comment by shankarpriyank** [comment]
i forgot adding a line to save the changes, should work as intended now.
in case it doesn't, please send me a couple of pictures with exif location

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
It is working great now!
https://commons.m.wikimedia.org/wiki/File%3ATOKUYAMA-brand_%E5%A1%A9%E5%8C%96%E3%82%AB%E3%83%AB%E3%82%B7%E3%82%A6%E3%83%A0_on_bridge_to_Koneshima.jpg

Would you mind fixing these 2 unit tests?

2023-10-18T14:50:39.7865366Z 
2023-10-18T14:50:39.7867579Z fr.free.nrw.commons.locationpicker.LocationPickerActivityUnitTests > testOnClickModifyLocation FAILED
2023-10-18T14:50:39.7870023Z     java.lang.reflect.InvocationTargetException at LocationPickerActivityUnitTests.kt:220
2023-10-18T14:50:39.7871412Z         Caused by: java.lang.NullPointerException at LocationPickerActivityUnitTests.kt:220
2023-10-18T14:50:40.2892517Z 
2023-10-18T14:50:40.2894112Z fr.free.nrw.commons.locationpicker.LocationPickerActivityUnitTests > testOnStyleLoaded FAILED
2023-10-18T14:50:40.2895823Z     java.lang.reflect.InvocationTargetException at LocationPickerActivityUnitTests.kt:204

**Comment by shankarpriyank** [comment]
I am having a hard time fixing the tests
`java.lang.instrument.IllegalClassFormatException: Error while instrumenting fr/free/nrw/commons/LocationPicker/LocationPickerActivity with JaCoCo 0.8.10.202304240956/8ea9668.
`
This the error in both of them, any idea how to fix them?

**Comment by nicolas-raoul** [comment]
Feel free to create a new issue about it asking for help, referring to your branch.

**Comment by rohit9625** [comment]
@shankarpriyank 
Did you create any other issue for the unit tests that required merging this PR?

**Pull request closed without merging**